internal/command/command.go

```diff
@@ -355,10 +356,246 @@ var CmdUpdateApis = &Command{
+			return err
+		}
+
+		pipelineState, err := loadState(languageRepo)
+		if err != nil {
+			fmt.Println("Error loading pipeline state:", err)
+			return err
+		}
+
+		librariesToRelease, prDescription, err := generateReleaseCommitForEachLibrary(ctx, repoPath, languageRepo, inputDirectory, pipelineState)
+		if err != nil {
+			return err
+		}
+
+		return generateReleasePr(ctx, librariesToRelease, pipelineState, repoPath, languageRepo, prDescription)
+	},
+}
+
+func setupLanguageAndInputFolders(ctx context.Context) (*gitrepo.Repo, string, string, error) {
+	startOfRun := time.Now()
+	tmpRoot, err := createTmpWorkingRoot(startOfRun)
+	var languageRepo *gitrepo.Repo
+	if flagRepoRoot == "" {
+		languageRepo, err = cloneLanguageRepo(ctx, flagLanguage, tmpRoot)
+		if err != nil {
+			return nil, "", "", err
+		}
+	} else {
+		languageRepo, err = gitrepo.Open(ctx, flagRepoRoot)
+		if err != nil {
+			return nil, "", "", err
+		}
+	}
+
+	inputDir := filepath.Join(tmpRoot, "inputs")
+	if err := os.Mkdir(inputDir, 0755); err != nil {
+		fmt.Println("Failed to create input directory:", err)
+		return nil, "", "", err
+	}
+
+	repoPath := flagRepoRoot
+	if repoPath == "" {
+		repoPath = filepath.Join(tmpRoot, fmt.Sprintf("google-cloud-%s", flagLanguage))
+	}
+
+	return languageRepo, repoPath, inputDir, nil
+}
+
+func checkFlags() error {
+	if !supportedLanguages[flagLanguage] {
+		return fmt.Errorf("invalid -language flag specified: %q", flagLanguage)
+	}
+	if flagImage == "" {
+		return fmt.Errorf("Create release pr requires image")
+	}
+	return nil
+}
+
+func generateReleasePr(ctx context.Context, librariesToRelease map[string]string, pipelineState *statepb.PipelineState, repoPath string, repo *gitrepo.Repo, prDescription string) error {
+	if len(librariesToRelease) > 0 {
+		updateLibraryMetadata(librariesToRelease, pipelineState, repoPath)
+		err := saveState(repo, pipelineState)
+		if err != nil {
+			return err
+		}
+		_, err = gitrepo.AddAll(ctx, repo)
+		if err != nil {
+
+		}
+		if err := gitrepo.Commit(ctx, repo, "updating pipeline-state with latest versions"); err != nil {
+			slog.Info(fmt.Sprintf("Received error trying to commit: '%s'", err))
+			return err
+		}
+
+		err = push(ctx, repo, time.Now(), "chore(main): release", "Release "+prDescription)
+		if err != nil {
+			slog.Info(fmt.Sprintf("Received error trying to create release PR: '%s'", err))
+			return err
+		}
+	}
+	return nil
+}
+
+/*
+this goes through each library in pipeline state and checks if any new commits have been added to that library since previous commit tag
+*/
+func generateReleaseCommitForEachLibrary(ctx context.Context, repoPath string, repo *gitrepo.Repo, inputDirectory string, pipelineState *statepb.PipelineState) (map[string]string, string, error) {
+	libraries := pipelineState.LibraryReleaseStates
+	var prDescription string
+	var librariesToRelease map[string]string
+	librariesToRelease = make(map[string]string)
+	for i := 0; i < len(libraries); i++ {
```

Comment:
Use range instead? `for _, library : range libraries`? (Ditto below for the source paths.)